Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide/Internal.hs

```diff
@@ -89,3 +111,7 @@ cancelRideTransaction booking ride bookingCReason merchantId rideEndedBy = do
   QBCR.upsert bookingCReason
   void $ QRB.updateStatus booking.id SRB.CANCELLED
   when (bookingCReason.source == SBCR.ByDriver) $ QDriverStats.updateIdleTime driverId
+  whenJust cancellationFee $ \fee -> do
+    riderId <- booking.riderId & fromMaybeM (BookingFieldNotPresent "rider_id")
```

Comment:
Can we avoid throw error and just do nothing if this case is happening?

Reply:
Sure, made change